visitor-app/src/visitors/pages/SignUp/signup_form.js

```diff
@@ -122,7 +123,7 @@ const signupForm = (props) => {
 
               <ErrorText show={props.errors.ageCheck}>{props.errors.ageCheck}</ErrorText>
             </div>
-            <SubmitButton type="submit">CONTINUE</SubmitButton>
+            <SubmitButton type="submit" disabled={props.status === status.PENDING}>CONTINUE</SubmitButton>
```

Comment:
Maybe we should also change the button text to "Please wait" or something like that when the request is pending?

Reply:
I think that will look jarring if there is a short request time and a validation error.

I can add a spinner internal to the button if you want. I started playing with one before. I was thinking a bounce loader would work:
https://www.react-spinners.com/

Reply:
Let's see what it looks like I guess. If it looks ok then let's go for it. Why did you think bounce loading instead of beat loader? The latter would make it seem more consistent with the other loaders?

Reply:
Options:

![button_1](https://user-images.githubusercontent.com/14337958/64690860-5305ae00-d489-11e9-8437-801a585bf523.gif)

![button_2](https://user-images.githubusercontent.com/14337958/64690869-5862f880-d489-11e9-9a22-f3433eb01dff.gif)

![button_3](https://user-images.githubusercontent.com/14337958/64690876-5d27ac80-d489-11e9-9b06-5a8cb68f7dac.gif)

![button_4](https://user-images.githubusercontent.com/14337958/64690884-61ec6080-d489-11e9-95fe-094422fcf146.gif)

Reply:
IMO the grey ones look better, leaning towards the first one (only one dot), but no strong opinion

Reply:
Opted for single loader in yellow as it complements the switch better:

![button_5](https://user-images.githubusercontent.com/14337958/64693915-4afd3c80-d490-11e9-91b2-0c90efb8c255.gif)